scripts/download-diff-metadata.ps1

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+
+Push-Location $env:BUILD_SOURCESDIRECTORY/msgraph-metadata
+
+function Format-Xml {
+    <#
+    .SYNOPSIS
+    Format the incoming object as the text of an XML document.
+    #>
+    param(
+        ## Text of an XML document.
+        [Parameter(ValueFromPipeline = $true)]
+        [string[]]$Text
+    )
+    
+    begin {
+        $data = New-Object System.Collections.ArrayList
+    }
+    process {
+        [void] $data.Add($Text -join "`n")
+    }
+    end {
+
+        $doc = New-Object System.Xml.XmlDocument
+        $doc.LoadXml($data -join "`n")
+        
+        $memStream = New-Object System.IO.MemoryStream
+        $writer = New-Object System.Xml.XmlTextWriter($memStream, [System.Text.Encoding]::UTF8)
+        $writer.Formatting = [System.Xml.Formatting]::Indented
+        $doc.WriteContentTo($writer)
+        $writer.Flush()
+        $memStream.Flush()
+
+        $memStream.Position = 0
+        $sReader = New-Object System.IO.StreamReader($memStream)
+
+        return $sReader.ReadToEnd()
+    }
+}
+
+# Are we on master? If not, we will want our changes committed on master.
+$branch = &git rev-parse --abbrev-ref HEAD
+Write-Host "Current branch: $branch"
+if ($branch -ne $env:targetBranch) {
+    git fetch origin $env:targetBranch | Write-Host
+    git checkout $env:targetBranch | Write-Host
+    $branch = &git rev-parse --abbrev-ref HEAD
+    Write-Host "Current branch: $branch"
+    git pull origin $env:targetBranch --ff-only | Write-Host
+}
```

Comment:
Aren't we in control of what branch we are on now? Would there be any scenario where we have a feature branch, and want to push into that particular branch?

Reply:
We always target master for the metadata. ADO checks out the repo detached HEAD state. I don't think we need feature branches here. This is old so perhaps there is better way. Do we even need to checkout master? Can we add and commit the changes in a detached HEAD state and push to origin/master? Any guidance is appreciated.

Reply:
I think this can be changed to 

`git checkout master`

Reply:
If Azure DevOps doesn't fetch all the branches, we may need to keep `git fetch origin master` before `git checkout master` as well,